lib/jmap/mail/email/email.dart

```diff
@@ -0,0 +1,112 @@
+import 'package:equatable/equatable.dart';
+import 'package:jmap_dart_client/http/converter/email_id_nullable_converter.dart';
+import 'package:jmap_dart_client/http/converter/email_id_converter.dart';
+import 'package:jmap_dart_client/http/converter/keyword_identifier_converter.dart';
+import 'package:jmap_dart_client/http/converter/mailbox_ids_converter.dart';
+import 'package:jmap_dart_client/http/converter/unsigned_int_nullable_converter.dart';
+import 'package:jmap_dart_client/http/converter/utc_date_nullable_converter.dart';
+import 'package:jmap_dart_client/jmap/mail/email/email_address.dart';
+import 'package:jmap_dart_client/jmap/mail/email/keyword_identifier.dart';
+import 'package:jmap_dart_client/jmap/core/id.dart';
+import 'package:jmap_dart_client/jmap/core/properties/properties.dart';
+import 'package:jmap_dart_client/jmap/core/unsigned_int.dart';
+import 'package:jmap_dart_client/jmap/core/utc_date.dart';
+import 'package:jmap_dart_client/jmap/mail/mailbox/mailbox.dart';
+
+class Email with EquatableMixin {
+  static Properties allProperties = Properties({
+    'id', 'threadId', 'mailboxIds', 'keywords', 'size', 'receivedAt',
+    'subject', 'sentAt', 'hasAttachment', 'preview', 'from', 'to'
+  });
+
+  final EmailId id;
+  final EmailId? threadId;
+  final Map<MailboxId, dynamic>? mailboxIds;
+  final Map<KeyWordIdentifier, dynamic>? keywords;
+  final UnsignedInt? size;
+  final UTCDate? receivedAt;
+  final String? subject;
+  final UTCDate? sentAt;
+  final bool? hasAttachment;
+  final String? preview;
+  final List<EmailAddress>? emailAddressSender;
+  final List<EmailAddress>? emailAddressReceiver;
+
+  Email(
+    this.id,
+    this.threadId,
+    this.mailboxIds,
+    this.keywords,
+    this.size,
+    this.receivedAt,
+    this.subject,
+    this.sentAt,
+    this.hasAttachment,
+    this.preview,
+    this.emailAddressSender,
+    this.emailAddressReceiver,
+  );
+
+  factory Email.fromJson(Map<String, dynamic> json) {
+    return Email(
+      const EmailIdConverter().fromJson(json['id'] as String),
+      const EmailIdNullableConverter().fromJson(json['threadId'] as String?),
+      json['mailboxIds'] == null
+          ? null
+          : (json['mailboxIds'] as Map<String, dynamic>).map((key, value) => MailboxIdsConverter().convert(key, value)),
+      json['mailboxIds'] == null
+          ? null
+          : (json['keywords'] as Map<String, dynamic>).map((key, value) => KeyWordIdentifierConverter().convert(key, value)),
```

Comment:
so I think we don't need this convert. And we can delegate this to `_$EmailFromJson`

Reply:
It contains an object that cannot be converted using JsonSerializable

Reply:
what object?

Reply:
you mean Map<...>, please try with build runner,

Reply:
> Could not generate `fromJson` code for `mailboxIds` because of type `MailboxId`.
Map keys must be one of: Object, dynamic, enum, String, BigInt, DateTime, int, Uri.

Ok, can not do it